session.go

```diff
@@ -106,17 +108,19 @@ func (s *session) close() error {
 }
 
 func (s *session) start(localMasterKey, localMasterSalt, remoteMasterKey, remoteMasterSalt []byte, profile ProtectionProfile, child streamSession) error {
-	var err error
-	s.localContext, err = CreateContext(localMasterKey, localMasterSalt, profile, s.localOptions...)
+	localContext, err := CreateContext(localMasterKey, localMasterSalt, profile, s.localOptions...)
```

Comment:
can we call `s.UpdateContext`? Just brings down the duplicated code (hopefully)

Reply:
You're absolutely right, plus it removes the need for the extra unit test